lib/gis/strlcpy.c

```diff
@@ -0,0 +1,55 @@
+/*!
+ * \file lib/gis/strlcpy.c
+ *
+ * \brief GIS Library - GRASS implementation of strlcpy().
+ *
+ * Loïc Bartoletti - 2024-07-25
+ *
+ * Copyright (c) 1998, 2015 Todd C. Miller <Todd.Miller@courtesan.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/*
+ * Copy string src to buffer dst of size dsize.  At most dsize-1
+ * chars will be copied.  Always NUL terminates (unless dsize == 0).
+ * Returns strlen(src); if retval >= dsize, truncation occurred.
+ */
```

Comment:
Please format this description to doxygen format, see eg. https://github.com/OSGeo/grass/blob/0306f9333d0ce45e32ee970af569e876c99918aa/lib/gis/asprintf.c#L26-L39, which results in [G_vasprintf()](https://grass.osgeo.org/programming8/asprintf_8c.html#aff45baf8dbedf969f66c8646fb9a5c54).

You should probably note that `src` must be NUL-terminated C string.

Reply:
Right. Done with your note about `src`